File: internship_backend_apr25/jury_solutions/cpp/b.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
#include <limits>

using namespace std;

int main ()
{
	string temp;
	getline(cin, temp);
	vector<int> houses;
	vector<int> heaters;

	int n;
	while (cin >> n)
	{
		houses.push_back(n);
		if (cin.peek() == '\n')
			break;
	}
	while (cin >> n)
	{
		heaters.push_back(n);
		if (cin.peek() == '\n')
			break;
	}

	vector<int> min_radius(houses.size(), numeric_limits<int>::max());

	int j = 0;
	int k = 0;

	sort(houses.begin(), houses.end());
	sort(heaters.begin(), heaters.end());

	while (j < houses.size())
	{
		min_radius[j] = min(min_radius[j], abs(houses[j] - heaters[k]));

		if (k > 0)
			min_radius[j] = min(min_radius[j], abs(houses[j] - heaters[k - 1]));

		if (heaters[k] < houses[j] && k < heaters.size() - 1)
			k++;
		else
			j++;
	}

	cout << *max_element(min_radius.begin(), min_radius.end()) << endl;

	return 0;
}

File: internship_backend_apr25/jury_solutions/cs/b.cs
using System;
using System.Linq;


public static class Program {
    public static void Main() {
    	var _ = Console.ReadLine();
    	var houses = Console.ReadLine().Split().Select(int.Parse).ToArray();
    	var heaters = Console.ReadLine().Split().Select(int.Parse).ToArray();
    
        var minRadius = new int[houses.Length];
        for (var i = 0; i < minRadius.Length; i++)
            minRadius[i] = int.MaxValue;
    
        var j = 0;
        var k = 0;

        Array.Sort(houses);
        Array.Sort(heaters);

        while (j < houses.Length ){
            minRadius[j] = Math.Min(minRadius[j], Math.Abs(houses[j] - heaters[k]));

            if (k > 0)
                minRadius[j] = Math.Min(minRadius[j], Math.Abs(houses[j] - heaters[k - 1]));

            if (heaters[k] < houses[j] && k < heaters.Length - 1)
                k++;
            else
                j++;
        }

        Console.WriteLine(minRadius.Max());
    }
}